flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskExecutorFileMergingManager.java

```diff
@@ -53,7 +57,8 @@ public class TaskExecutorFileMergingManager {
      * manager(executor).
      */
     @GuardedBy("lock")
-    private final Map<JobID, FileMergingSnapshotManager> fileMergingSnapshotManagerByJobId;
```

Comment:
I'm not sure if using <JobID, ExecutionAttemptID> directly as the key is enough? to avoid nesting.

Reply:
Well I suggest no. We still need to call `releaseMergingSnapshotManagerForJob` in `releaseJobResources` as a safety guard, where we use the job id to retrieve a manager.